lib/server.js

```diff
@@ -150,33 +161,36 @@ var routes = [
     }
   },
 
-  {
+  { // slack retrigger command
     url: /^\/slack.*/,
     handler: function(request, respond, onError) {
-      var patchId = request.text;
-      var fragments = decodeURIComponent(request.url).
-        match(/[\/?&]text=([^&]+)/)[1].
-        split(" ");
-
-      var jobName = fragments[0];
-      var buildId = fragments[1];
-
-      var jobLink = [ K.JENKINS_URL, 'job', jobName, buildId ].join("/");
+      var jobLink = slackUrlToJenkins(request.url)
 
       jenkins.connect().then(function() {
         return jenkins.retrigger(jobLink);
       }).then(function() {
         return jenkins.getJob(jobLink);
-      }).then(function(jobStatus) {
-        respond(200, jobStatus);
+      }).then(function() {
+        respond(200, "Your job has been retriggered.");
+      }, onError);
+    }
+  },
+
+  { // slack job status command
+    url: /^\/jobstatus.*/,
```

Comment:
I'm not sure I'm following here. We already have an endpoint for retrieving job's status, `/job?link=...`, how is this different? Is it that you don't know the URL/link and you want the server to generate it?

If that's the case, please add a different query parameter to the existing endpoint that would cause the server to generate the link you want, i.e. `text`.